From a65a43e46b9662f429369736134945c9af6f0f00 Mon Sep 17 00:00:00 2001
From: Greg Kaiser <gkaiser@google.com>
Date: Wed, 1 Aug 2018 12:34:32 -0700
Subject: [PATCH 7/8] Revert "Support Speck encryption."

This reverts commit 49c27c5cb2c48ce4665affe2b3086077c768a9ae.

Remove the Speck encryption support. It was eventually
decided not to allow Speck in Android P, so this code
is no longer needed and wasn't used outside of testing.

Bug: 112009351
Test: Confirmed AES continues to work with FBE.
Change-Id: Ia5458143be5687fff8d541d8fa2c8ee24a369da4
(cherry picked from commit 270e90f033dd412206697e3a27aee76c24038c33)

# Conflicts:
#	fs_mgr/fs_mgr_fstab.cpp
---
 fs_mgr/fs_mgr_fstab.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/fs_mgr/fs_mgr_fstab.cpp b/fs_mgr/fs_mgr_fstab.cpp
index 0677e3d4b..8fc04c865 100644
--- a/fs_mgr/fs_mgr_fstab.cpp
+++ b/fs_mgr/fs_mgr_fstab.cpp
@@ -114,13 +114,10 @@ static struct flag_list fs_mgr_flags[] = {
 #define EM_ICE          2
 #define EM_AES_256_CTS  3
 #define EM_AES_256_HEH  4
-#define EM_SPECK_128_256_XTS 5
-#define EM_SPECK_128_256_CTS 6
 #define EM_ICE_WRAPPED_KEY_SUPPORTED 7
 
 static const struct flag_list file_contents_encryption_modes[] = {
     {"aes-256-xts", EM_AES_256_XTS},
-    {"speck128/256-xts", EM_SPECK_128_256_XTS},
     {"software", EM_AES_256_XTS}, /* alias for backwards compatibility */
     {"ice", EM_ICE},              /* hardware-specific inline cryptographic engine */
     {"ice_wrapped_key_supported", EM_ICE_WRAPPED_KEY_SUPPORTED}, /* ICE engine with wrapped key support */
@@ -130,7 +127,6 @@ static const struct flag_list file_contents_encryption_modes[] = {
 static const struct flag_list file_names_encryption_modes[] = {
     {"aes-256-cts", EM_AES_256_CTS},
     {"aes-256-heh", EM_AES_256_HEH},
-    {"speck128/256-cts", EM_SPECK_128_256_CTS},
     {0, 0},
 };
 
-- 
2.17.1

